### DISCUSSION TITLE
Ideas for additional procfs collection

### DISCUSSION THREAD

**Discussion by halpomeranz** [tclahr/uac#34] [Ideas]
live_response/process/procfs_information.yaml is a good group of items to collect, but I would suggest adding a few more items-- at least on Linux and Linux-like systems:

- "ls -l /proc/[0-9]*/cwd" is one of my go-to items for detecting suspicious processes-- when the CWD is /tmp/.ICEd-unix/fooTWUX67 you know you have a problem
- "cat /proc/%line%/stack" can sometimes reveal details of process behavior-- e.g., waiting on a socket, etc
- "cat /proc/%line%/status" has lots of extra process detail, including PPID etc

Also if /proc/&lt;pid&gt;/exe or any of /proc/&lt;pid&gt;/fd/* show up as being "(deleted)", it would be cool if UAC could have a command-line switch that would enable automatically grabbing copies of these files.

**Comment by tclahr** [comment]
hi Hal,

I will update live_response/process/procfs_information.yaml as you suggested (3 bullets).

The last suggestion about deleted binaries (/proc/<pid>/exe and /proc/<pid>/fd/*) can also be added. The drawback would be UAC output file being flagged as "malicious" if it is transferred to a workstation running antivirus software.

Creating a password-protected zip file could be a solution on systems where zip tool is available. Any suggestions?

Thank you for the suggestions!

**Reply by halpomeranz** [reply]
There are multiple external programs that could be used to encode files: zip, gzip, openssl, "dd conv=swab", "xxd -p", and so on. Maybe UAC could just use whatever is available.

An alternative would be to implement a simple XOR encoder in UAC itself.

**Reply by tclahr** [reply]
Yeah, I will check if there is a common tool I can use across all the supported OSs. If there is none, I will implement a simple XOR encoder as you suggested.

Thank you!

**Comment by tclahr** [comment]
Please refer to issues #35 and #36

**Comment by tclahr** [comment]
I have update UAC to collects copies of '/proc/<pid>/exe' and their related '/proc/<pid>/fd/*' if they are shown up as being (deleted). They are copied using 'dd conv=swab' tool in order to avoid UAC output file being flagged and quarantined by any antivirus tool.

Clone the following branch if you want to test it please -> https://github.com/tclahr/uac/tree/feature/issue-36
